spot/spot_code/Generate_Spot_Annotations.py

```diff
@@ -0,0 +1,260 @@
+        for b in tqdm(self.barcodes):
+
+            # Pulling out that row from coordinates
+            b_coords = self.coordinates.loc[b]
+            b_x = b_coords['imagecol']
+            b_y = b_coords['imagerow']
+            
+            # Creating shapely annotation for this spot
+            spot_poly = Point(b_x, b_y).buffer(spot_pixel_radius)
+
+            # Extracting -omics information from previous processing step
+            spot_main_cell_types = self.omics_data['main_cell_types'].loc[b].to_frame()
+            spot_main_cell_types.columns = ['Main_Cell_Types']
+            spot_main_cell_types_dict = spot_main_cell_types.to_dict()
+
+            # Cell State info
+            spot_cell_state_dict = {}
+            spot_cell_subtype_dict = {}
+            for m in list(spot_main_cell_types_dict['Main_Cell_Types'].keys()):
+
+                spot_cell_state = self.omics_data[m]['pct_states'][b].to_frame()
+                spot_cell_state.columns = [m]
+                spot_cell_state_dict[m] = spot_cell_state.to_dict()[m]
+
+                spot_cell_subs = self.omics_data[m]['pct_subtypes'][b].to_frame()
+                spot_cell_subs.columns = [m]
+                spot_cell_subs_dict = spot_cell_subs.to_dict()
+                spot_cell_subtype_dict[m] = spot_cell_subs_dict[m]
+
+            spot_properties = {
+                'Main_Cell_Types': spot_main_cell_types_dict['Main_Cell_Types'],
+                'Cell_States': spot_cell_state_dict,
+                'Cell_Subtypes': spot_cell_subtype_dict
+                }
+            # Adding spot to annotations, crs is just the origin since we are using non-scaled centroid points
+            # name is set to be the barcode
+            spot_annotations.add_shape(
+                poly=spot_poly,
+                box_crs=[0, 0],
+                structure='Spots',
+                name=b,
+                properties=spot_properties)
+
+        # Print str property of Annotation object
+        print(spot_annotations)
+        return spot_annotations
+
+    def save(self):
+
+        os.makedirs(self.args.basedir + '/tmp/')
+        f = open(self.args.basedir + '/tmp/spotanot.json', 'w')
+        f.close()
+        self.spot_annotations.json_save(self.args.basedir + '/tmp/spotanot.json')
+        annot = json.load(open(self.args.basedir + '/tmp/spotanot.json'))
+        folder = self.args.basedir
+        girder_folder_id = folder.split('/')[-2]
+        _ = os.system("printf 'Using data from girder_client Folder: {}\n'".format(folder))
+        file_name = self.args.input_files.split('/')[-1]
+        gc = girder_client.GirderClient(apiUrl=self.args.girderApiUrl)
+        gc.setToken(self.args.girderToken)
+        files = list(gc.listItem(girder_folder_id))
+        item_dict = dict()
+        for file in files:
+            d = {file['name']: file['_id']}
+            item_dict.update(d)
+        print(item_dict)
+        print(item_dict[file_name])
```

Comment:
Can the item id just be included in the input arguments? It shouldn't cause too much of a problem but it seems unnecessary.

Reply:
In the Histomics UI user will not have access to itemId, he needs to go back and copy the value. And also all the other plugins are implemented like this.  So I can change if you want to.